machine.go

```diff
@@ -196,7 +205,9 @@ func NewMachine(ctx context.Context, cfg Config, opts ...Opt) (*Machine, error)
 		return nil, err
 	}
 
-	m := &Machine{}
+	m := &Machine{
+		exitCh: make(chan struct{}),
```

Comment:
Why not have this be an `chan error`? On a successful flow of the `Machine` you simply could just the close the channel and any error can be sent through it.The reason why is the introduction of the `err` field seems unneeded if we use this and allows for multiple errors to be sent down the channel if we wanted.

Reply:
A channel has no persistence; once you read an item out of the channel, it's gone.  To support `Wait` returning the same error to every caller, we need to store it in a variable instead of a channel.

Reply:
My biggest concern is potentially stomping on top of pre-existing errors set on `m.err`. If we could make it impossible to stomp over errors, I'd be more in favor of this.

Reply:
`m.err` is written exactly once, on line 355 inside `startVMM`.  The behavior of `Start` only being called once should make this reasonably safe, but I could make it safer with another `sync.Once` in `startVMM` if you're concerned about it.

Reply:
Naw, don't think we need another `sync.Once` for this. We had something similar in the AWS SDK for Go which was on requests. Requests also had a `Copy` method, but we would still stomp over the errors from time to time which made debugging some issues difficult. I am okay with taking this as is for now since the error isn't exposed which will allow us to change it if we can think of a better method of addressing of the stomping over errors.